src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -498,7 +565,7 @@ private static Exception InvalidTypeException(object value, string argument)
         }
 
         [Serializable]
-        private class Enumerator : IEnumerator<T>, IEnumerator
+        private struct Enumerator : IEnumerator<T>, IEnumerator
```

Comment:
Why did you change this to be a struct? It will not provide any performance gains because it is handed out as an `IEnumerator<T>` during `GetEnumerator`, so now it'll need to be boxed.